include/boost/http_proto/service/impl/zlib_service.hpp

```diff
@@ -0,0 +1,72 @@
+//
+// Copyright (c) 2021 Vinnie Falco (vinnie.falco@gmail.com)
+// Copyright (c) 2024 Mohammad Nejati
+//
+// Distributed under the Boost Software License, Version 1.0. (See accompanying
+// file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
+//
+// Official repository: https://github.com/cppalliance/http_proto
+//
+
+#ifndef BOOST_HTTP_PROTO_SERVICE_IMPL_ZLIB_SERVICE_HPP
+#define BOOST_HTTP_PROTO_SERVICE_IMPL_ZLIB_SERVICE_HPP
+
+#include <boost/system/is_error_code_enum.hpp>
+
+namespace boost {
+
+namespace system {
+template<>
+struct is_error_code_enum<
+    ::boost::http_proto::zlib::error>
+{
+    static bool const value = true;
+};
+} // system
+
+namespace http_proto {
+namespace zlib {
+
+namespace detail {
+
+struct BOOST_SYMBOL_VISIBLE
+    error_cat_type
```

Comment:
I didn't realize URL put so much in the headers. Is this the best technique? Maybe... I can't remember why though.